src/main/java/gyro/google/compute/HealthCheckFinder.java

```diff
@@ -0,0 +1,150 @@
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.HealthCheck;
+import com.google.api.services.compute.model.HealthCheckList;
+import gyro.core.GyroException;
+import gyro.core.Type;
+import gyro.google.GoogleFinder;
+
+/**
+ * Query for health checks.
+ *
+ * ========
+ * Examples
+ * ========
+ *
+ * Example find all global health checks for project.
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    compute-health-check: $(external-query google::compute-health-check)
+ *
+ * Example find the global health check on name equal to 'health-check-example'..
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     compute-health-check: $(external-query google::compute-health-check { name: "health-check-example" })
+ *
+ * Example find all the health checks for the region 'us-east1'.
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     compute-health-check: $(external-query google::compute-health-check { region: "us-east1" })
+ *
+ */
+@Type("compute-health-check")
+public class HealthCheckFinder extends GoogleFinder<Compute, HealthCheck, HealthCheckResource> {
+
+    private String name;
+    private String region;
+
+    /**
+     * The name of the health check.
+     */
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The name of the region for this request. Not applicable to global health checks..
+     */
+    public String getRegion() {
+        return region;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    @Override
+    protected List<HealthCheck> findAllGoogle(Compute client) {
+        try {
+            List<HealthCheck> healthChecks = new ArrayList<>();
+            HealthCheckList healthCheckList;
+            String nextPageToken = null;
+            do {
+                healthCheckList = client.healthChecks().list(getProjectId()).setPageToken(nextPageToken).execute();
+                healthChecks.addAll(healthCheckList.getItems());
+                nextPageToken = healthCheckList.getNextPageToken();
+            } while (nextPageToken != null);
+            return healthChecks;
+        } catch (GoogleJsonResponseException je) {
+            throw new GyroException(je.getDetails().getMessage());
+        } catch (IOException ex) {
+            throw new GyroException(ex);
+        }
+    }
+
+    @Override
+    protected List<HealthCheck> findGoogle(Compute client, Map<String, String> filters) {
+        try {
+            List<HealthCheck> healthChecks = new ArrayList<>();
+            HealthCheckList healthCheckList;
+            String nextPageToken = null;
+
+            if (filters.containsKey("region")) {
+                do {
+                    healthCheckList = client.regionHealthChecks()
+                        .list(getProjectId(), filters.get("region"))
```

Comment:
before calling the `list` method, is `name` present in the filter should be checked, and if it is then the `get` method should be fired.